File: cs-ssbu-project/src/getEventId.js
require('dotenv').config();
const fetch = (...args) => import('node-fetch').then(({ default: fetch }) => fetch(...args));

const startggURL = "https://api.start.gg/gql/alpha";
const startggKey = process.env.REACT_APP_STARTGG_API_KEY;

//console.log('Start.gg API Key:', startggKey); // Agrega esta línea para verificar la clave

if (!startggKey) {
    throw new Error('La clave de la API de Start.gg no está configurada. Verifica tu archivo .env');
}

const getEventId = async (tournamentName, eventName) => {
    const eventSlug = `tournament/${tournamentName}/event/${eventName}`;

    try {
        const response = await fetch(startggURL, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
                'Accept': 'application/json',
                'Authorization': 'Bearer ' + startggKey
            },
            body: JSON.stringify({
                query: "query EventQuery($slug: String) { event(slug: $slug) { id name } }",
                variables: { slug: eventSlug }
            })
        });

        const data = await response.json();
        console.log('API response:', data);

        if (data.errors) {
            console.error('API returned errors:', data.errors);
            throw new Error('Error in API response');
        }

        if (!data.data || !data.data.event) {
            console.error('Response structure:', data);
            throw new Error('Invalid response structure');
        }

        const eventId = data.data.event.id;
        console.log('Event ID:', eventId);
        return eventId;

    } catch (error) {
        console.error('Error al obtener el ID del evento:', error);
        throw error;
    }
};

module.exports = { getEventId };

File: cs-ssbu-project/backend/CSSSBUBD/myapp/api/getEventId.py
import os
import requests
from dotenv import load_dotenv
from django.http import JsonResponse

# Cargar la clave de la API desde el archivo .env
load_dotenv()

startgg_url = "https://api.start.gg/gql/alpha"
startgg_key = os.getenv('REACT_APP_STARTGG_API_KEY')

if not startgg_key:
    raise ValueError('La clave de la API de Start.gg no está configurada. Verifica tu archivo .env')

def get_event_id(tournament_name, event_name):
    event_slug = f"tournament/{tournament_name}/event/{event_name}"
    
    headers = {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        'Authorization': f'Bearer {startgg_key}'
    }
    
    query = """
    query EventQuery($slug: String) {
        event(slug: $slug) {
            id
            name
        }
    }
    """
    
    variables = {
        'slug': event_slug
    }
    
    try:
        response = requests.post(
            startgg_url,
            headers=headers,
            json={'query': query, 'variables': variables}
        )
        
        response.raise_for_status()
        
        data = response.json()
        print('API response:', data)

        if 'errors' in data:
            print('API returned errors:', data['errors'])
            return JsonResponse({"error": "Error in API response"}, status=400)

        if 'data' not in data or 'event' not in data['data']:
            print('Response structure:', data)
            return JsonResponse({"error": "Invalid response structure"}, status=400)

        event_id = data['data']['event']['id']
        return event_id
     
    except requests.exceptions.RequestException as e:
        print('Error al obtener el ID del evento:', e)
        raise e

# Vista para manejar la solicitud desde el frontend
def get_event_id_view(request):
    tournament_name = request.GET.get('tournament_name')
    event_name = request.GET.get('event_name')

    if not tournament_name or not event_name:
        return JsonResponse({"error": "Missing parameters"}, status=400)

    try:
        event_id = get_event_id(tournament_name, event_name)
        return JsonResponse({"event_id": event_id}, status=200)
    except Exception as e:
        return JsonResponse({"error": str(e)}, status=500)
